app/src/main/java/com/example/playlistmaker/SearchActivity.kt

```diff
@@ -45,13 +66,78 @@ class SearchActivity : AppCompatActivity() {
             val inputMethodManager = getSystemService(Context.INPUT_METHOD_SERVICE) as? InputMethodManager
             inputMethodManager?.hideSoftInputFromWindow(currentFocus!!.windowToken, 0)
             editText.clearFocus()
+            tracks.clear()
+            adapter.notifyDataSetChanged()
+        }
+
+        updateButton.setOnClickListener {
+            findTracks()
         }
 
         editText.doOnTextChanged { s, start, before, count ->
             searchText = editText.text.toString()
             clearButton.visibility = clearButtonVisibility(s)
         }
 
+        editText.setOnEditorActionListener { _, actionId, _ ->
+            if (actionId == EditorInfo.IME_ACTION_DONE) {
+                if (editText.text.isNotEmpty()) {
+                    findTracks()
+                }
+                true
+            }
+            false
+        }
+
+    }
+
+    @SuppressLint("NotifyDataSetChanged")
+    private fun showMessage(text: String) {
+        if (text == "nothing found") {
+            notInternet.visibility = View.GONE
+            nothingFound.visibility = View.VISIBLE
+            tracks.clear()
+            adapter.notifyDataSetChanged()
+
+        } else if (text == "not internet"){
+            nothingFound.visibility = View.GONE
+            notInternet.visibility = View.VISIBLE
+            tracks.clear()
+            adapter.notifyDataSetChanged()
+        } else {
+            nothingFound.visibility = View.GONE
+            notInternet.visibility = View.GONE
+        }
+    }
+
+    private fun findTracks() {
+        iTunesSearchService.search(editText.text.toString()).enqueue(object :
```

Comment:
```suggestion
    private fun findTracks(query: String) {
        iTunesSearchService.search(query).enqueue(object :
```
Лучше передавать в эту функцию строку, по которой необходимо выполнить поисковый запрос. Это поможет тебе исправить замечание, связанное с кнопкой "Обновить"